js/coinone.js

```diff
@@ -371,78 +371,253 @@ module.exports = class coinone extends Exchange {
+        const result = [];
+        const completOrders = this.safeValue (response, 'completeOrders');
+        for (let i = 0; i < completOrders.length; i++) {
+            const order = completOrders[i];
+            const timestamp = this.safeTimestamp (order, 'timestamp');
+            let side = this.safeString (order, 'type');
+            let feeCurrency = undefined;
+            if (side.indexOf ('ask') >= 0) {
+                side = 'sell';
+                feeCurrency = market['quote'];
+            } else {
+                side = 'buy';
+                feeCurrency = market['base'];
+            }
+            const price = this.safeFloat (order, 'price');
+            const amount = this.safeFloat (order, 'qty');
+            let cost = undefined;
+            if (price !== undefined && amount !== undefined) {
+                cost = price * amount;
+            }
+            result.push ({
+                'id': this.safeString (order, 'orderId'),
+                'clientOrderId': undefined,
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+                'lastTradeTimestamp': undefined,
+                'status': 'closed',     // might be 'canceled'
+                'symbol': market['symbol'],
+                'type': 'limit',
+                'side': side,
+                'price': price,
+                'amount': amount,
+                'filled': amount,    // coinone set amount as same as filled for completed order
+                'remaining': 0,      // amount is always equal to filled
+                'cost': cost,
+                'trades': undefined,    // coinone doesn't support fills
+                'fee': {
+                    'currency': feeCurrency,
+                    'rate': this.safeFloat (order, 'feeRate'),
+                    'cost': this.safeFloat (order, 'fee'),
+                },
+                'info': order,
+            });
+        }
+        return result;
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        // [WARNING] Because of the limited capabilities of Coinone V2 API, you should be noticed below
+        // * The symbol parameter must be given. If not, an exception will occur.
+        // * The since and limit parameters are totally ignored.
+        // * At maximum, 50 of my recent trades can be returned.
+        await this.loadMarkets ();
+        if (symbol === undefined) {
+            throw new ExchangeError (this.id + ' allows fetching my trades with a specific symbol');
+        }
+        const market = this.market (symbol);
+        const request = {
+            'currency': market['id'],
+        };
+        const response = await this.privatePostOrderCompleteOrders (this.extend (request, params));
+        const result = [];
+        const completOrders = this.safeValue (response, 'completeOrders');
+        for (let i = 0; i < completOrders.length; i++) {
```

Comment:
We need to move this parsing section to `parseTrade()`, to keep it in line with almost all other exchanges.

Reply:
To my mind it is not entirely correct, since a trade is not the same as a closed order.

Reply:
Coinone DOES NOT provide namely 'my trades' api.

By the way, **privatePostOrderCompleteOrders() responses my trades**. If an order has multiple fills, all of the fills are responded by privatePostOrderCompleteOrders(), each by each. This is true event if the order is open.

Reply:
The exchange can name their endpoints how they want. Some exchanges can have an "orders" endpoint, actually serving the fills, not orders.  And vice versa. 

So, in general case, we should not rely on the naming on the exchange side. Instead, we should look into the returned structures. If those structures are trades – we should implement `fetchMyTrades () { ... privatePostOrderCompleteOrders() ... }`, however, if it returns orders – we should implement `fetchClosedOrders() { ... privatePostOrderCompleteOrders() ... }`. 

The main criteria is: CCXT's fetchClosedOrders() should not return multiple orders with the same order id. It should only return separate standalone orders (not related to each other). If the structures in the array have the same order id repeated multiple times – those are trades. Otherwise to a user that will look like multiple orders with the same id, but all other fields differing from one order to another. Which would be a problem for the user.

If you can post a `privatePostOrderCompleteOrders` verbose output – that would really help us get on track.

Reply:
Your main criteria is totally agreed.

My suggestion is:
In fetchClosedOrders(), the trades from privatePostOrderCompleteOrders() can be aggregated by order id.

And here is an actual response from privatePostOrderCompleteOrders(). You can find 2 occurrences of id '50ff2638-1e4d-11e9-9ec7-00e04c3600d7' :

```json
{
    "errorCode": "0",
    "completeOrders": [
        {
            "orderId": "50ff4ee0-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "4.0487",
            "timestamp": "1591275107",
            "price": "3624.0",
            "qty": "0.5586",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50ff4ee0-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "16.1978",
            "timestamp": "1591275107",
            "price": "3624.0",
            "qty": "2.2348",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50ff3088-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "3.627",
            "timestamp": "1591274344",
            "price": "3627.0",
            "qty": "0.5",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50ff32a7-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.001",
            "timestamp": "1591274344",
            "price": "3627.0",
            "qty": "0.5",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50ff2638-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.00052",
            "timestamp": "1591274243",
            "price": "3620.0",
            "qty": "0.26",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50ff2638-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.00208",
            "timestamp": "1591274243",
            "price": "3620.0",
            "qty": "1.04",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50ff18b6-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.002002",
            "timestamp": "1591274156",
            "price": "3616.0",
            "qty": "1.001",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50ff1cc9-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "3.6087",
            "timestamp": "1591274155",
            "price": "3616.0",
            "qty": "0.499",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50ff18b6-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.000998",
            "timestamp": "1591274155",
            "price": "3616.0",
            "qty": "0.499",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50ff0f6e-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.001",
            "timestamp": "1591274054",
            "price": "3623.0",
            "qty": "0.5",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50d62fa9-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "3.4281",
            "timestamp": "1591203205",
            "price": "3435.0",
            "qty": "0.499",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50d43cd6-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "2.041",
            "timestamp": "1591185843",
            "price": "3420.0",
            "qty": "0.2984",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50d43cd6-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "3.42",
            "timestamp": "1591185806",
            "price": "3420.0",
            "qty": "0.5",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50d46933-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.001",
            "timestamp": "1591185806",
            "price": "3420.0",
            "qty": "0.5",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50d3eac3-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "3.4071",
            "timestamp": "1591185402",
            "price": "3414.0",
            "qty": "0.499",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50d40bc6-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.0016",
            "timestamp": "1591185351",
            "price": "3410.0",
            "qty": "0.8",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50d3e84a-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.001",
            "timestamp": "1591184735",
            "price": "3415.0",
            "qty": "0.5",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "50d38a0e-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "1.3606",
            "timestamp": "1591183975",
            "price": "3405.0",
            "qty": "0.1998",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "50d37242-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.0004",
            "timestamp": "1591183686",
            "price": "3397.0",
            "qty": "0.2",
            "feeRate": "0.002",
            "type": "bid"
        },
        {
            "orderId": "5095f191-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "6.7447",
            "timestamp": "1591061698",
            "price": "3369.0",
            "qty": "1.001",
            "feeRate": "0.002",
            "type": "ask"
        },
        {
            "orderId": "4f50f2d5-1e4d-11e9-9ec7-00e04c3600d7",
            "fee": "0.002005",
            "timestamp": "1590408177",
            "price": "3069.0",
            "qty": "1.0025",
            "feeRate": "0.002",
            "type": "bid"
        }],
    "result": "success"
}
```

Reply:
@smcho Thanks!

> My suggestion is:
> In fetchClosedOrders(), the trades from privatePostOrderCompleteOrders() can be aggregated by order id.

Unfortunately, we cannot really do that, because that implies that all trades are on the same page. If it so happens that some of them "go out of range" (slide to a second page) – we will end up with a wrong aggregation. We could loop over pages, but there's no definitive stopping criteria for such a loop, and that break the policy of "one request per unified call". So it's "either we fetch all pages or nothing" in a general case. However, running such a loop inside a unified call will make it very hard for the user to control the rate limits, especially when running async unified calls.

We've had similar difficulties with other exchanges, therefore we refrain from such emulation, since the user may get confused in this case. 

I would just treat those as `fetchMyTrades`, without having a `fetchClosedOrders`.

Let me know your thoughts. Thx again for your help!

Reply:
> I would just treat those as fetchMyTrades, without having a fetchClosedOrders. That would reflect the actual API truth to the user properly – those are just trades, not orders.

I agree. API consistency over various exchanges is very very important for CCXT users.

Please, delete fetchClosedOrders() from my PR.

Thank you.

Reply:
Will do, and will merge shortly, thank you, I appreciate your involvement!